android/src/main/java/io/sentry/react/RNSentryModule.java

```diff
@@ -343,15 +345,29 @@ public void captureScreenshot(Promise promise) {
             return;
         }
 
-        final byte[] raw = takeScreenshot(activity, logger, buildInfo);
-        if (raw == null) {
+        CountDownLatch doneSignal = new CountDownLatch(1);
+        final byte[][] bytesWrapper = {{}};
+        UiThreadUtil.runOnUiThread(() -> {
+            bytesWrapper[0] = takeScreenshot(activity, logger, buildInfo);
+            doneSignal.countDown();
+        });
+
+        try {
+            doneSignal.await();
```

Comment:
M: Should we set a default await time? just wondering if this can be stuck somehow.

Reply:
I don't know, but better be safe, I'll add a timeout.